src/benchmarklib/abstract_table_generator.cpp

```diff
@@ -348,10 +348,11 @@ void AbstractTableGenerator::generate_and_store() {
   Hyrise::get().set_scheduler(initial_scheduler);
 }
 
-std::shared_ptr<BenchmarkConfig> AbstractTableGenerator::create_benchmark_config_with_chunk_size(
-    ChunkOffset chunk_size) {
+std::shared_ptr<BenchmarkConfig> AbstractTableGenerator::create_benchmark_config_with_chunk_size(ChunkOffset chunk_size,
```

Comment:
I don't like that change too much, given the name now only covers one of the two things that this function handles.
Also not a fan of default parameters.

But for  now, I don't have a better idea.

Reply:
We could also not use the convencience constructors of the TableGenerators, but create the benchmark config in the console and call the regular constructors from there.

Reply:
I guess I'd prefer that because this function is kind of strange anyways?!?

Reply:
I think that's because `BenchmarkConfig::get_default_config()` returns an object and not a shared ptr, which is almost always required, and that wrapper does that. Just searched the codebase and basically everywhere where `get_default_config()` is called, a shared ptr using the copy constructor is created afterwards.

Reply:
Change the interface then?

Reply:
Also my first thought. But still, the following is not as easily possible w/o some utility wrapping the assignment of a custom chunk size:
https://github.com/hyrise/hyrise/blob/804f9ed10d7ee197868614287ec5403b3728c205/src/benchmarklib/tpch/tpch_table_generator.cpp#L129-L131

And apparently, we use at least TPCHTableGenerator's convenience constructor at multiple places. Removing the constructor could be one option. Then, a convenience creation method for a BenchmarkConfig could help to not write
```c++
const auto config = ...
config->chunk_size = chunk_size
...TableGenerator{..., config}.generate_and_store();
```
instead of
```c++
...TableGenerator{..., chunk_size}.generate_and_store();  // current way
```
or
```c++
...TableGenerator{..., BenchmarkConfig::with_chunk_size(chunk_size)}.generate_and_store();  // one option
```

Such a function could also be used by the table generators' convenience constructors if we do not want to modify their interfaces. And then we would have sth like:
```c++
// Returns shared_ptr instead of object.
std::shared_ptr<BenchmarkConfig> BenchmarkConfig::create_default_config() {
  return std::make_shared<BenchmarkConfig>(); 
}

// Naming tbd.
std::shared_ptr<BenchmarkConfig> BenchmarkConfig::with_chunk_size(const ChunkOffset chunk_size) {
  auto config = create_default_config();
  config->chunk_size = chunk_size;
  return config; 
}
```

Idk, what's your take on that?

Reply:
I think the simplest solution would be my favorite. Having a few selected constructors for the benchmark config and use `make_shared` with these instead of having specialized functions in the abstract table generator.

But now that I see the result `...TableGenerator{..., std::make_shared<BenchmarkConfig>(chunk_size)}.generate_and_store();`, I guess I agree with your proposal of the convenience functions. But within the BenchmarkConfig itself and I would rather have overloads than named functions. What do you think?

Reply:
Yeah, I guess overload constructors for `BenchmarkConfig` also sound viable.

Reply:
I added overloaded constructors, and thus, got also rid of `BenchmarkConfig::get_default_config()`. @Bouncner do you wanna give it a look?